services/book.test.js

```diff
@@ -0,0 +1,60 @@
+const bookRepository = require('../repositories/books');
+const bookService = require('./book');
+
+jest.mock('../repositories/books');
+
+describe('Book Service', () => {
+  describe('#getNotFoundBooks', () => {
+    test('should call bookRepository.getAllBooks once', async () => {
+      try {
+        await bookService.getNotFoundBooks(['some book title']);
+      } catch {}
+
+      expect(bookRepository.getAllBooks).toHaveBeenCalledTimes(1);
+    });
+
+    test('should throw an error when bookRepository.getAllBooks throws an error', async () => {
+      bookRepository.getAllBooks = jest.fn().mockImplementation(() => {
+        throw new Error('books not found');
+      });
+
+      try {
+        await bookService.getNotFoundBooks(['some book title']);
+      } catch (error) {
+        expect(error).toStrictEqual(Error('books not found'));
```

Comment:
this line is a bit tricky: it'll pass both when error is thrown and when is not. 
to avoid such a behaviour you can declare a variable outside, assign in catch block an assert after

Reply:
Thanks for checking this. I think i found a better way to write this using: https://jestjs.io/docs/expect#rejects. I'm going to try it out.